ticker/simpleMock.go

```diff
@@ -0,0 +1,58 @@
+package ticker
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+// SimpleMock implements the Ticker interface, and provides a method of
+// force-feeding ticks, even while paused.
+type SimpleMock struct {
+	IsActive uint32 // used atomically
+
+	// Force is used to force-feed a ticks into the ticker. Useful for
+	// debugging when trying to wake an event.
+	Force chan time.Time
+}
+
+// NewSimpleMock returns a SimpleMock Ticker, used for testing and debugging. It
+// supports the ability to force-feed events that get output by the
+func NewSimpleMock() *SimpleMock {
+	m := &SimpleMock{
+		Force: make(chan time.Time),
+	}
+	return m
+}
+
+// Ticks returns a receive-only channel that delivers times at the ticker's
+// prescribed interval when active. Force-fed ticks can be delivered at any
+// time.
+//
+// NOTE: Part of the Ticker interface.
+func (m *SimpleMock) Ticks() <-chan time.Time {
+	return m.Force
+}
+
+// Resume starts underlying time.Ticker and causes the ticker to begin
```

Comment:
Since this is to be _purely_ force fed, do we need the resumption logic?

Reply:
I left it in so we can assert whether the ticker is paused/running at the right times. I also made a 
new change to not expose the Force chan directly. It needs to be force-ticked through a func now which also checks isActive.